cpp/ql/lib/semmle/code/cpp/controlflow/IRGuards.qll

```diff
@@ -1007,24 +1107,25 @@ private predicate compares_lt(
   exists(boolean isGe | isLt = isGe.booleanNot() | compares_ge(test, left, right, k, isGe, value))
   or
   /* (x is true => (left < right + k)) => (!x is false => (left < right + k)) */
-  exists(AbstractValue dual | value = dual.getDualValue() |
-    compares_lt(test.(LogicalNotInstruction).getUnary(), left, right, k, isLt, dual)
+  exists(AbstractValue dual |
+    value = dual.getDualValue() and
+    compares_lt(test.(LogicalNotValueNumber).getUnary(), left, right, k, isLt, dual)
```

Comment:
Why the switch from `exists( ... | ... | ...)` to `exists( ... | ... and ...)`?

Reply:
Oh, that's a mistake. I think I was quick-evaling `... and ...` at some point during development since it's still not possible to quick eval `... | ...` 😭

I don't have a strong preference on `... | ...` vs. `... and ...` in this case since my normal motivation for preferring `... | ...` over `... and ...` is to save parentheses 😄

Reply:
Let's just change this back to `exists( ... | ... | ...)` then.

Reply:
Fixed in c80bea33e527fe2a149c2cf54f2a23138eef370c